### PR TITLE
add properties for IoT Hub's routing example

### PR DESCRIPTION
**Pull request by ggthename**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x ] Other... Please describe:
```
According to https://docs.microsoft.com/ko-kr/azure/iot-hub/tutorial-routing, I'm trying to implement routing example.
But there is only C# simulation code.
So, I added a property "level" for this tutorial.

## How to Test
*  Get the code

git clone https://github.com/ggthename/azure-iot-samples-node
cd iot-hub/Tutorials/Routing
npm install

* Test the code

npm install
node SimulatedDevice.js "{your device connection string}"

## What to Check
Verify that the following are valid
* https://docs.microsoft.com/ko-kr/azure/iot-hub/tutorial-routing
You can see exactly same result.

### CODE REVIEW

File: iot-hub/Tutorials/ConnectivityTests/SimulatedDevice-3.js [hunk 1]
```diff
@@ -82,6 +82,16 @@ client.open(function (err) {
       var data = JSON.stringify({ temperature: temperature, humidity: humidity });
       var message = new Message(data);
 
+      // Add the level
+      var level = Math.random();
+      if(level >= 0.7){
+        message.properties.level="critical"
```
Comment (pierreca) [inline iot-hub/Tutorials/ConnectivityTests/SimulatedDevice-3.js]: I don't think this is a valid use of the properties` object of the `Message` class. see https://github.com/Azure/azure-iot-sdk-node/blob/master/common/core/src/properties.ts for details. I'd be surprised if this worked.

Also, as a nitpick: spacing of braces/curly braces is inconsistent with other samples.

**Comment by ggthename** [comment]
Thanks for your opinion. That was my mistake. I'll fix it

**Comment by dominicbetts** [comment]
Hi @ggthename - thanks for the contribution. Your proposed changes are to a sample that relates to a different tutorial. Can you instead create a new sample (by copying an existing one) with the Message properties and place it in a folder called iot-hub/Tutorials/Routing to be consistent with our naming convention.

**Comment by dominicbetts** [comment]
Additionally, your commit should not include the .DS_store or package_lock.json files. Thanks!

**Comment by ggthename** [comment]
Thank you! @dominicbetts. I'm sorry to make lots of commit :)

File: iot-hub/Tutorials/Routing/SimulatedDevice.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+// Connect to the IoT hub.
+client.open(function (err) {
+  if (err) {
+    console.log(chalk.red('Could not connect: ' + err));
+  } else {
+    console.log('Client connected');
+
+    // Listen for TestMethod being called from the hub
+    client.onDeviceMethod('TestMethod', function(request, response) {
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/SimulatedDevice.js]: The equivalent C# sample doesn't listen for method calls - you can remove this.

File: iot-hub/Tutorials/Routing/SimulatedDevice.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        if(err) {
+          console.error(chalk.red('An error ocurred when sending a method response:\n' + err.toString()));
+        } else {
+          console.log(chalk.green('Response to method \'' + request.methodName + '\' sent successfully.' ));
+        }
+      });
+    });
+
+    // Send reported properties to the hub
+    client.getTwin(function(err, twin) {
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/SimulatedDevice.js]: The equivalent C# sample doesn't use twins - you can remove this.

File: iot-hub/Tutorials/Routing/SimulatedDevice.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      var temperature = 20 + (Math.random() * 15);
+      var humidity = 60 + (Math.random() * 20);
+
+      // Add the telemetry to the message body.
+      var data = JSON.stringify({ temperature: temperature, humidity: humidity });
+      var message = new Message(data);
+
+      // Add the level for routing
+      var level = Math.random();
+      if (level >= 0.7) {
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/SimulatedDevice.js]: The logic you have here isn't exactly the same as the C# sample - the C# sample uses two random numbers.

File: iot-hub/Tutorials/Routing/SimulatedDevice.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    });
+
+    // Create a message and send it to the IoT hub every second
+    setInterval(function(){
+      // Simulate telemetry.
+      var temperature = 20 + (Math.random() * 15);
+      var humidity = 60 + (Math.random() * 20);
+
+      // Add the telemetry to the message body.
+      var data = JSON.stringify({ temperature: temperature, humidity: humidity });
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/SimulatedDevice.js]: The equivalent C# sample also adds the device id and an information string to the message payload. Can you add these?

File: iot-hub/Tutorials/Routing/SimulatedDevice.js (hunk 2 above)
Comment (ggthename) [reply iot-hub/Tutorials/Routing/SimulatedDevice.js]: sure!

File: iot-hub/Tutorials/Routing/SimulatedDevice.js (hunk 5 above)
Comment (ggthename) [reply iot-hub/Tutorials/Routing/SimulatedDevice.js]: Sure. I added these values to the message payload. Could you check it?

File: iot-hub/Tutorials/Routing/SimulatedDevice.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+// The sample connects to a device-specific MQTT endpoint on your IoT Hub.
+var Protocol = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+var NoRetry = require('azure-iot-common').NoRetry;
+var client = Client.fromConnectionString(connectionString, Protocol);
+const chalk = require('chalk');
+const myDeviceId = 'Contoso-Test-Device';
+
+console.log('IoT Hub troubleshooting tutorial\nSimulated device #3\n')
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/SimulatedDevice.js]: This line should be something like:
`console.log('IoT Hub routing tutorial\nSimulated device\n')`

File: iot-hub/Tutorials/Routing/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "simulated-device",
+  "version": "1.0.0",
+  "description": "Simulated device application in the IoT Hub connectivity tutorial",
```
Comment (dominicbetts) [inline iot-hub/Tutorials/Routing/package.json]: routing tutorial

**Comment by dominicbetts** [comment]
@pierreca - this looks good to me now from the docs perspective.

**Comment by ggthename** [comment]
please let me know when it's going on.

**Comment by ggthename** [comment]
please let me know when it's going on....(2)

**Comment by ryanwinter** [comment]
Closing as repository is being archived. 

Please create a PR on [azure-iot-sdk-node](https://github.com/Azure/azure-iot-sdk-node) if you see value in this PR.

**Pull request closed without merging**